tests/checks/integration/test_snmp.py

```diff
@@ -107,19 +113,58 @@ class SNMPTestCase(AgentCheckTest):
         }
     ]
 
-    CHECK_TAGS = ['snmp_device:localhost']
+    def run_check(self, config, agent_config=None, mocks=None, force_reload=False):
+        if force_reload and self.check:
+            self.check.stop()
+        super(SNMPTestCase, self).run_check(config, agent_config, mocks, force_reload)
+
+    def tearDown(self):
+        if self.check:
+            self.check.stop()
 
     @classmethod
     def generate_instance_config(cls, metrics):
         instance_config = copy.copy(cls.SNMP_CONF)
         instance_config['metrics'] = metrics
+        instance_config['name'] = "localhost"
         return instance_config
 
+    def wait_for_async(self, method, attribute, count):
+        """
+        Loop on `self.check.method` until `self.check.attribute >= count`.
+
+        Raise after
+        """
+        i = 0
+        while i < RESULTS_TIMEOUT:
+            self.check._process_results()
+            if len(getattr(self.check, attribute)) >= count:
+                return getattr(self.check, method)()
+            time.sleep(1)
+            i += 1
+        raise Exception("Didn't get the right count of service checks in time, {0}/{1} in {2}s: {3}"
```

Comment:
The method is generic, not the exception message mentioning "the right count of service checks"